mne/viz/topo.py

```diff
@@ -226,24 +240,49 @@ def format_coord_multiaxis(x, y, ch_name=None):
     if unified:
         under_ax._mne_axs = axs
         # Create a PolyCollection for the axis backgrounds
+        sel_pos = pos[[i[0] for i in iter_ch]]
         verts = np.transpose(
             [
-                pos[:, :2],
-                pos[:, :2] + pos[:, 2:] * [1, 0],
-                pos[:, :2] + pos[:, 2:],
-                pos[:, :2] + pos[:, 2:] * [0, 1],
+                sel_pos[:, :2],
+                sel_pos[:, :2] + sel_pos[:, 2:] * [1, 0],
+                sel_pos[:, :2] + sel_pos[:, 2:],
+                sel_pos[:, :2] + sel_pos[:, 2:] * [0, 1],
             ],
             [1, 0, 2],
         )
-        if not img:
-            under_ax.add_collection(
-                collections.PolyCollection(
-                    verts,
-                    facecolor=axis_facecolor,
-                    edgecolor=axis_spinecolor,
-                    linewidth=1.0,
+        if not img:  # Not needed for image plots.
+            collection = collections.PolyCollection(
+                verts,
+                facecolor=axis_facecolor,
+                edgecolor=axis_spinecolor,
+                linewidth=1.0,
+            )
+            under_ax.add_collection(collection)
+
+            if select:
+                # Configure the lasso-selection tool
+                fig.lasso = SelectFromCollection(
+                    ax=under_ax,
+                    collection=collection,
+                    names=shown_ch_names,
+                    alpha_nonselected=0,
+                    alpha_selected=1,
+                    linewidth_nonselected=0,
+                    linewidth_selected=0.7,
                 )
-            )  # Not needed for image plots.
+
+                def on_select():
+                    publish(fig, ChannelsSelect(ch_names=fig.lasso.selection))
+
+                def on_channels_select(event):
+                    ch_inds = {name: i for i, name in enumerate(shown_ch_names)}
+                    selection_inds = [
+                        ch_inds[name] for name in event.ch_names if name in ch_inds
+                    ]
```

Comment:
I think this is equivalent and clearer... if you agree that it's clearer code, please double-check the equivalence!

```suggestion
                    selection_inds = list(
                        np.nonzero(np.isin(shown_ch_names, event.ch_names))[0]
                    )
```

Reply:
have to disagree with you on this one. Using two functions and a magic `[0]` is not clearer IMO

Reply:
do as you will.  reasonable folks will disagree on many judgments of style / clarity

Reply:
I now got it better. Thanks for pointing it out to me.